File: backend/src/main/java/com/app/backend/dao/entities/RendezVous.java
package com.app.backend.dao.entities;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import java.sql.Time;


import java.util.Date;

@Document(collection = "rdv")
@Data
@AllArgsConstructor
@NoArgsConstructor
@ToString
@Getter
@Setter
@Builder
public class RendezVous {
    @Id
    private String id;
    //date demande de RDV
    private Date dateDemande;
    //date demandée pour le RDV
    private Date dateRDV;
    private Time heureRDV;
    private boolean valide;
    private boolean vu;
    @DBRef
    private User patient;
    @DBRef
    private User medecin;

}

File: frontend/src/app/shared/RendezVous/rendez-vous.ts
import { User } from "../user";

export class RendezVous {
  id: string;
  dateDemande: Date;
  date: Date;
  heure: string; // Use string or a custom Time type
  valide: boolean;
  vu: boolean;
  patient: User;
  medecin: User;

  constructor(
    id: string,
    dateDemande: Date,
    date: Date,
    heure: string,
    valide: boolean,
    vu: boolean,
    patient: User,
    medecin: User
  ) {
    this.id = id;
    this.dateDemande = dateDemande;
    this.date = date;
    this.heure = heure;
    this.valide = valide;
    this.vu = vu;
    this.patient = patient;
    this.medecin = medecin;
  }
}
